src/web3-adapter/customRPC.ts

```diff
@@ -0,0 +1,27 @@
+// web3-core-method exports both a function and a type called `Method`. Typescript defaults to importing the type rather than the function, so when I try to call it typescript will yell at me. So we use require instead.
+/* tslint:disable-next-line */
+const Method = require("web3-core-method");
+
+interface PatchParams {
+  name: string;
+  call: string;
+  params: [any];
+  inputFormatter: any;
+  outputFormatter: any;
+}
+
+export function patchEnableCustomRPC(web3: any): void {
+  web3.eth.customRPC = function (opts: PatchParams) {
+    /* tslint:disable-next-line */
+    const self = this;
```

Comment:
I wrote the following before I realized that this code is copied from the original web3.js source, so you can ignore it. But since I went through the trouble to type it, I'm posting it anyways.

This line doesn't serve any functional purpose, unless you just like having the variable named `self` instead of `this`.

The place where you most commonly see this sort of thing is when you want to capture `this` in a callback function, for example:
```ts
function addMyselfAsListener() {
  const self = this;
  button.onClick = function() {
    self.doTheThing();
  };
}
```
In this case, not capturing `this` in a separate variable leads to the dreaded `this`-binding issues where when the callback is called, it assigns `this` based on the context where the callback is called (probably setting it to `undefined`). But people don't usually do this anymore, because you get the exact same behavior by using an arrow-function instead:
```ts
function addMyselfAsListener() {
  button.onClick = () => this.doTheThing();
}
```

Reply:
This is a good point, I just removed the line as it's completely unnecessary. Thanks for being thorough!